sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/sampleoverview/SampleView.groovy

```diff
@@ -2,141 +2,209 @@ package life.qbic.portal.sampletracking.components.sampleoverview
+        return presenter
+    }
+
+    private static String determineColor(Status status) {
+        switch (status) {
+            case Status.DATA_AVAILABLE:
+                return State.COMPLETED.getCssClass()
+            case Status.SAMPLE_QC_FAIL:
+                return State.FAILED.getCssClass()
+            default:
+                return State.IN_PROGRESS.getCssClass()
+        }
+    }
+
+    private void addColumnColoring() {
+        sampleGrid.getColumn("status").setStyleGenerator({ Sample sample -> determineColor(sample.status) })
+    }
+
+    void enableUserSampleFiltering() {
+        enableUserFilterByStatus()
+        enableUserFilterBySearchbar()
+    }
+
+    private void enableUserFilterByStatus() {
+        ComboBox<Status> statusComboBox = this.statusComboBox
+
+        statusComboBox.setItems(Status.METADATA_REGISTERED,
+                Status.SAMPLE_RECEIVED,
+                Status.SAMPLE_QC_FAIL,
+                Status.SAMPLE_QC_PASS,
+                Status.LIBRARY_PREP_FINISHED,
+                Status.DATA_AVAILABLE)
+        statusComboBox.setItemCaptionGenerator({ it.getDisplayName() })
+
+        DataProvider<Sample, ?> dataProvider = this.sampleGrid.getDataProvider()
+        statusComboBox.addValueChangeListener({
+            if (it.getValue()) {
+                sampleFilter.withStatus(it.getValue().toString())
+            } else {
+                sampleFilter.clearStatus()
+            }
+            dataProvider.refreshAll()
+        })
+    }
+
+    void enableUserFilterBySearchbar() {
+        TextField searchField = this.searchField
+        DataProvider<Sample, ?> dataProvider = this.sampleGrid.getDataProvider()
+        searchField.addValueChangeListener({
+            if (it.getValue()) {
+                sampleFilter.containingText(it.getValue())
+            } else {
+                sampleFilter.containingText("")
+            }
+            dataProvider.refreshAll()
+        })
+    }
+
+    private void setSampleGridStyle() {
+        setHeaderRowStyle()
+    }
+
+    private void setHeaderRowStyle() {
+        HeaderRow headerRow = sampleGrid.getDefaultHeaderRow()
+        headerRow.getCell("name").setStyleName("cell-min-width")
+        headerRow.getCell("code").setStyleName("cell-min-width")
+        headerRow.getCell("status").setStyleName("cell-min-width")
+    }
 
-    Presenter(NotificationService notificationService, ViewModel viewModel) {
-      this.notificationService = notificationService
-      this.viewModel = viewModel
+    /**
+     * Adds responsiveness to an abstractComponent
+     *
+     * <p>This applies the css class style .responsive-grid-layout to the provided abstractComponent allowing it to display it's content in a responsive manner</p>
+     *
+     * @param AbstractComponent the {@link com.vaadin.ui.AbstractComponent}, where the css style and responsiveness should be added
+     * @since 1.0.2
+     */
+    void setupLayoutResponsiveness() {
```

Comment:
Same as above, why is a setup method public? If this is an requirement for the view to work properly, I suggest to call it internally during construction.